File: easy/Unary/OCaml/Unary.ml
(* Function to convert an integer to a 7-bit binary string *)
let to_binary n =
  let rec aux n acc =
    if n = 0 then acc
    else aux (n / 2) ((string_of_int (n mod 2)) ^ acc)
  in
  let binary = aux n "" in
  String.make (7 - String.length binary) '0' ^ binary (* Pad with leading zeros *)

let () =
  let msg = input_line stdin in

  let output = ref "" in
  let bn = ref "" in
  let b = ref '\000' in (* Track the previous bit *)

  (* Convert each character to its binary representation *)
  for i = 0 to String.length msg - 1 do
    let c = msg.[i] in
    (* Convert to ASCII *)
    let ascii = Char.code c in
    (* Convert to binary and pad to 7 bits *)
    let binary = to_binary ascii in
    bn := !bn ^ binary;
  done;

  (* Convert the binary string to unary *)
  for i = 0 to String.length !bn - 1 do
    let c = String.get !bn i in
    if c <> !b then begin
      if c = '0' then
        output := !output ^ " 00 "
      else
        output := !output ^ " 0 ";
      b := c;
    end;
    output := !output ^ "0";
  done;

  (* Trim the leading space *)
  let trimmed_output = String.trim !output in
  print_endline trimmed_output

File: easy/Unary/Pascal/Unary.pas
program Answer;
{$H+}
uses sysutils;

var
    MESSAGE: String;
    output: String;
    bn: String;
    b: Char;
    i, ascii: Integer;
    binary_value, binary: String;

// Function to convert an integer to a 7-bit binary string
function ToBinary(n: Integer): String;
var
    binary: String;
begin
    binary := '';
    while n > 0 do
    begin
        binary := IntToStr(n mod 2) + binary;
        n := n div 2;
    end;
    ToBinary := StringOfChar('0', 7 - Length(binary)) + binary; // Pad with leading zeros
end;

begin
    readln(MESSAGE);
    output := '';
    bn := '';
    b := #0; // Initialize the previous bit to null character

    // Convert each character to its binary representation
    for i := 1 to Length(MESSAGE) do
    begin
        ascii := Ord(MESSAGE[i]); // Convert character to ASCII
        binary_value := ToBinary(ascii); // Convert ASCII to binary
        bn := bn + binary_value; // Append the binary value to the binary string
    end;

    // Convert the binary string to unary
    for i := 1 to Length(bn) do
    begin
        if bn[i] <> b then
        begin
            if bn[i] = '1' then
                output := output + ' 0 '
            else
                output := output + ' 00 ';
            b := bn[i]; // Update the previous bit
        end;
        output := output + '0'; // Add a unary '0' for each bit
    end;

    // Trim leading spaces and print the result
    writeln(Trim(output));
end.